oc-game-swift/Heroes/Hero.swift

```diff
@@ -62,19 +62,75 @@ class Hero {
         if isDead {
             return "\(name) || \(type) 💀 DIED IN COMBAT 💀"
         } else {
-            return "\(name) || \(type) -- \(life)/\(maxLife)HP -- doing \(damage) DMG"
+            return "\(name) || \(type) -- \(life)/\(maxLife)HP -- doing \(weapon.damage) DMG"
         }
     }
     
     /// Attacking the hero whos targeting
     /// - target: Hero
     func attackHero(_ target: Hero) {
         // Amount of damage
-        target.life -= self.damage
+        target.life -= self.weapon.damage
         
         // If the target is dead
         if target.life <= 0 {
             target.isDead = true
         }
     }
+    
+    /// Switching weapon from the random bonus chest
+    func switchWeapon() {
+        // For the healer
+        if let hero = self as? Healer {
```

Comment:
maybe you should replace `if let` with a `switch...case...let`

Reply:
Moreover, maybe you could think about a shorter method for gathering all your conditions together ? I let you think about it (I have no solution for now)

Reply:
Done